src/magic_folder/snapshot.py

```diff
@@ -144,40 +154,64 @@ def tahoe_create_snapshot_dir(nodeurl, content, parents, timestamp, treq):
 class TahoeSnapshot(object):
     """
     Represents a snapshot corresponding to a file.
+
+    XXX we want a 'file name' of some sort .. is that relative to the
+    magic-folder base, or absolute, or that weird 'flattened' thing
+    magic-folder does?
     """
 
-    filepath = attr.ib()
-    node_directory = attr.ib(converter=lambda p: p.asBytesMode())
+    capability = attr.ib()
+    parents = attr.ib()
 
-    @inlineCallbacks
-    def create_snapshot(self, parents, treq):
-        """
-        Create a snapshot.
 
-        :param [unicode] parents: List of parent snapshots of the current snapshot
-            (read-caps of parent snapshots)
+@inlineCallbacks
+def create_snapshot(node_directory, filepath, parents, treq):
+    """
+    Create a snapshot.
```

Comment:
Yeah (and see comment above .. more in the "idea" stage here). I also don't know that we need a `node_directory` (instead, a WebUI endpoint? URL? maybe just whatever is behind `treq` can already know that?) and `filepath` should probably be a file-like, etc.

Reply:
> maybe just whatever is behind treq can already know that?) 

yea.  I think bundling up all of the tahoe-lafs web api information behind a single interface would be good, rather than passing several values around to any API that wants to make a webapi call.